bridges/snowbridge/pallets/xcm-helper/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+
+		/// The runtime `Origin` type.
+		type RuntimeOrigin: From<Origin> + From<<Self as frame_system::Config>::RuntimeOrigin>;
+
+		/// The runtime `Call` type.
+		type RuntimeCall: Parameter
+			+ GetDispatchInfo
+			+ Dispatchable<
+				RuntimeOrigin = <Self as Config>::RuntimeOrigin,
+				PostInfo = PostDispatchInfo,
+			>;
+
+		/// Means of measuring the weight consumed by an XCM message locally.
+		type Weigher: WeightBounds<<Self as Config>::RuntimeCall>;
+
+		/// Universal location of this runtime.
+		type UniversalLocation: Get<InteriorLocation>;
+
+		/// Ethereum's location of this runtime.
+		type Destination: Get<Location>;
+	}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Fees were paid from a location for an operation (often for using `SendXcm`).
+		FeesPaid { paying: Location, fees: Assets },
+		/// Execution of an XCM message was attempted.
+		Attempted { outcome: Outcome },
+		/// A XCM message was sent.
+		Sent { origin: Location, destination: Location, message: Xcm<()>, message_id: XcmHash },
+	}
+
+	#[pallet::origin]
+	#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+	pub enum Origin {
+		/// It comes from somewhere in the XCM space wanting to transact.
+		Xcm(Location),
+	}
+	impl From<Location> for Origin {
+		fn from(location: Location) -> Origin {
+			Origin::Xcm(location)
+		}
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		InvalidXcm,
+		SendFailure,
+		BadVersion,
+		Empty,
+		CannotReanchor,
+		CannotDetermine,
+		InvalidAsset,
+		FeesNotMet,
+		UnweighableMessage,
+		LocalExecutionIncomplete,
+	}
+
+	#[derive(Clone, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+	pub struct TransactInfo {
+		pub target: H160,
+		pub call: Vec<u8>,
+		pub gas_limit: u64,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight(Weight::from_parts(100_000_000, 0))]
+		pub fn transact_to_ethereum(
+			origin: OriginFor<T>,
+			target: H160,
+			call: Vec<u8>,
+			gas_limit: u64,
+		) -> DispatchResult {
+			let origin = T::ExecuteXcmOrigin::ensure_origin(origin)?;
+
+			let dest = T::Destination::get();
+
+			// construct the inner xcm of ExportMessage
+			let transact = TransactInfo { target, call, gas_limit };
+			let message = Xcm(vec![Transact {
+				origin_kind: OriginKind::SovereignAccount,
+				require_weight_at_most: Weight::default(),
+				call: transact.encode().into(),
+			}]);
+
+			Self::send_xcm(origin, dest, message, None)?;
+
+			Ok(())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight(Weight::from_parts(100_000_000, 0))]
+		pub fn transfer_to_ethereum(
+			origin: OriginFor<T>,
+			beneficiary: H160,
+			asset: Box<VersionedAsset>,
+			fee: Box<VersionedAsset>,
```

Comment:
Basically just mirror the `reserve_transfer_assets` extrinsic from `pallet_xcm` with an explicit fee specified here.

Reply:
Nice 🚀